src/library_pthread.js

```diff
@@ -272,7 +267,6 @@ var LibraryPThread = {
           } else {
             err('Internal error! Worker sent a message "' + cmd + '" to target pthread ' + d['targetThread'] + ', but that thread no longer exists!');
           }
-          PThread.currentProxiedOperationCallerThread = undefined;
```

Comment:
Is there a good place to clear `PThread.currentProxiedOperationCallerThread` in the new code?

Reply:
I'm not sure there is a lot of benefit from clearing this.. and its kind of tricky because the current code does:

```
return func.apply(null, emscripten_receive_on_main_thread_js_callArgs)
```

Changing that to the following seem unnecessary adds code size:

```
var rtn = func.apply(null, emscripten_receive_on_main_thread_js_callArgs);
PThread.currentProxiedOperationCallerThread = undefined;
return rtn;
```

Reply:
Ultimately I think we should find a better way to transmit this information to the receiver since this use a global variable is kind of gross and doesn't compose/nest.